storage/namespace.go

```diff
@@ -66,7 +67,8 @@ type dbNamespace struct {
 	blockRetriever block.DatabaseBlockRetriever
 	opts           Options
 	runtimeOptsMgr m3dbruntime.OptionsManager
-	nopts          namespace.Options
+	metadata       namespace.Metadata
+	seriesOpts     series.Options
```

Comment:
hm seems strange to have a series options here, conceptually namespaces only supposed to know about shards and shards contain series. Having the series options here couples namespaces and series directly.